specification/deviceupdate/data-plane/Microsoft.DeviceUpdate/stable/2022-10-01/deviceupdate.json

```diff
@@ -0,0 +1,4472 @@
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/IfNoneMatchParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Operation status.",
+            "headers": {
+              "Retry-After": {
+                "type": "string",
+                "description": "Number of seconds to wait before checking the operation status again."
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/DeviceOperation"
+            }
+          },
+          "304": {
+            "description": "Not modified."
+          },
+          "default": {
+            "description": "Default response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/deviceUpdate/{instanceId}/management/operations": {
+      "get": {
+        "tags": [
+          "Device and deployment management"
+        ],
+        "operationId": "DeviceManagement_ListOperations",
+        "description": "Get a list of all device import operations. Completed operations are kept for 7 days before auto-deleted.",
+        "x-ms-examples": {
+          "DeviceManagement_ListOperations": {
+            "$ref": "./examples/DeviceManagement_ListOperations.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/InstanceIdParameter"
+          },
+          {
+            "name": "filter",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "Restricts the set of operations returned. Only one specific filter is supported: \"status eq 'NotStarted' or status eq 'Running'\""
+          },
+          {
+            "name": "top",
+            "in": "query",
+            "required": false,
+            "type": "integer",
+            "format": "int32",
+            "description": "Specifies a non-negative integer n that limits the number of items returned from a collection. The service returns the number of available items up to but not greater than the specified value n."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Operation details.",
+            "schema": {
+              "$ref": "#/definitions/DeviceOperationsList"
+            }
+          },
+          "default": {
+            "description": "Default response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-odata": "#/definitions/OperationFilter",
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/deviceUpdate/{instanceId}/management/deviceDiagnostics/logCollections/{operationId}": {
```

Comment:
Let's just use "logCollectionId" instead of "operationId" here. Clearly that is the better name since you are remapping it to "logCollectionId" for the SDKs.

Reply:
We agree with the intent, but it does not meet the bar for GA.
This would result in unplanned for our API and a breaking change for our UX. 

Justification for won't fix:
1.	This is an issue of inconsistency – no functional impact.
2.	Diagnostics API is not part of our core scenarios.
3.	We haven’t received any customer feedback that this is a problem

Reply:
I think you should reconsider. GA is _the last time_ you could ever make this change. Once it is GA then you are stuck with it _forever_.

Reply:
I'll second this by adding that from a developer experience perspective, we *have* heard feedback from customers that the many little inconsistencies in Azure APIs cause confusion and unnecessary friction. Before it's stuck like this in GA, we should consider addressing it at the source.

Diagnostics APIs may not be a core scenario from a service perspective but it certainly is for the end-user, who is trying to understand what went wrong with their deployment.

Reply:
> I think you should reconsider. GA is _the last time_ you could ever make this change. Once it is GA then you are stuck with it _forever_.

Until deprecated. Understood.

Reply:
We don't deprecate APIs in Azure unless there is a compelling business need, because deprecating APIs means customers are negatively impacted.